walk-in-interview/src/main/webapp/account/firebase-auth.js

```diff
@@ -0,0 +1,240 @@
+ * file that wishes to use firebase auth:
+ * <script src="https://www.gstatic.com/firebasejs/7.17.1/firebase-app.js"></script>
+ * <script src="https://www.gstatic.com/firebasejs/7.17.1/firebase-auth.js"></script>
+ */
+
+import {AppStrings} from '../strings.en.js';
+import {setCookie, getCookie,
+  USER_TYPE_COOKIE_PARAM, USER_TYPE_NO_USER, USER_TYPE_APPLICANT} from '../common-functions.js';
+import {API} from '../apis.js';
+
+const firebaseConfig = {
+  apiKey: 'AIzaSyDhpzKNLAMNyEdw6ovQ5sPvnOhXDwhse-o',
+  authDomain: 'com-walk-in-interview.firebaseapp.com',
+  databaseURL: 'https://com-walk-in-interview.firebaseio.com',
+  projectId: 'google.com:walk-in-interview',
+  storageBucket: 'undefined',
+  messagingSenderId: '610715446188',
+  appId: '1:610715446188:web:8cec0c2798940bb9d32809',
+  measurementId: 'G-RJ55G4ZXN8',
+};
+
+firebase.initializeApp(firebaseConfig);
+
+// TODO(issue/21): get the language from the browser
+const CurrentLocale = 'en';
+
+firebase.auth().languageCode = CurrentLocale;
+
+// Firebase UI
+const ui = new firebaseui.auth.AuthUI(firebase.auth());
+
+const STRINGS = AppStrings['auth'];
+
+const Auth = {};
+
+/**
+ * This will create a new business account.
+ *
+ * @param {String} email The email for the new business account.
+ * @param {String} password The password for the new business account.
+ */
+Auth.createBusinessAccount = (email, password) => {
+  firebase.auth().createUserWithEmailAndPassword(email, password)
+      .catch((error) => {
+        console.error(error);
+      });
+};
+
+/**
+ * This will sign into an existing business account.
+ *
+ * @param {String} email The email for the exisiting business account.
+ * @param {String} password The password for the existing business account.
+ */
+Auth.signIntoBusinessAccount = (email, password) => {
+  return firebase.auth().signInWithEmailAndPassword(email, password);
+};
+
+/**
+ * This will add the firebase ui for phone authentication
+ * to the provided element.
+ *
+ * @param {String} elementId The div element to add the UI.
+ * @param {String} successPath The url for redirect on login success.
+ * @param {String} newUserInfo The message to be displayed if it is a new user.
+ */
+Auth.addPhoneSignInAndSignUpUI = (elementId, successPath, newUserInfo) => {
+  ui.start(`#${elementId}`, {
+    signInOptions: [
+      {
+        provider: firebase.auth.PhoneAuthProvider.PROVIDER_ID,
+        defaultCountry: 'SG',
+      },
+    ],
+    callbacks: {
+      signInSuccessWithAuthResult: (authResult, redirectUrl) => {
+        // TODO(issue/89): add new user pages for name/skills
+        // for now, this will only redirect to homepage for exisiting users
+
+        // TODO(issue/100): set the cookie at the server side instead
+        setCookie(USER_TYPE_COOKIE_PARAM, USER_TYPE_APPLICANT);
+
+        if (authResult.additionalUserInfo.isNewUser) {
+          // Informs user
+          alert(newUserInfo);
```

Comment:
//TODO(issue/) replace with proper notification <- same for all the other alerts.